### PR TITLE
Reduce geoprocessor calls

### PR DESCRIPTION
**Pull request by jtarricone**
This is a minor refactor that reduces the number of calls to the geoprocessor by bundling the AoI census into the JSON blob returned by the `analyze` endpoint, and subsequently using it when instantiating projects and scenarios so that it can be POSTed to the API modeling endpoints. The durations of the blocking geoprocessor calls are not trivial, especially for large AoI's.

To test:
* clone the branch and bring up the environment
* navigate to `http://localhost:8000`
* go through typical user actions, e.g. select an AoI for analysis, create a TR55 model, add some scenarios
* keeping the test version of the app open, switch branches to `develop` or navigate the to the staging site and perform the same user actions, trying to keep all the parameters as close to identical as possible
* compare the two to make sure the output is identical, or if the parameters of the model are slightly different, that the differences can reasonably be ignored
* in general, the test version should be faster than the baseline, especially if the AoI is somewhat large (>10k sq-km)

Connects #956

### CODE REVIEW

File: src/mmw/js/src/modeling/controllers.js [hunk 1]
```diff
@@ -94,7 +94,7 @@ var ModelingController = {
 
                     App.currentProject = project;
                     if (!App.projectNumber) {
-                        setupNewProjectScenarios(project);
+                        setupNewProjectScenarios(project, project.get('aoi_census'));
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Since the information is contained within `project`, we don't need to send `project.get('aoi_census')` as a second parameter here, we can just send `project` and have `setupNewProjectScenarios` extract it. Unless there are cases when this is supplied from outside `project`?

**Comment by kellyi** [comment]
`./scripts/debugcelery.sh` is showing an error when trying to Analyze this AOI:

screen shot 2016-09-27 at 3 46 25 pm

Here's the error:

```sh
[2016-09-27 15:46:22,842: ERROR/MainProcess] Task apps.modeling.tasks.histogram_to_survey_census[a3b72e39-0a3d-4e35-94f1-c5360c56b151] raised unexpected: TypeError('histogram_to_survey_census() takes exactly 1 argument (2 given)',)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
TypeError: histogram_to_survey_census() takes exactly 1 argument (2 given)
[2016-09-27 15:46:22,847: ERROR/Worker-1] Task a3b72e39-0a3d-4e35-94f1-c5360c56b151 run from job 1596 raised exception: histogram_to_survey_census() takes exactly 1 argument (2 given)
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
    return self.run(*args, **kwargs)
TypeError: histogram_to_survey_census() takes exactly 1 argument (2 given)
```

**Comment by kellyi** [comment]
(Looks like I'm encountering the same area for other AOIs, too.)

File: src/mmw/apps/modeling/tasks.py [hunk 2]
```diff
@@ -108,17 +108,18 @@ def get_histogram_job_results(self, incoming):
 
 
 @shared_task
-def histogram_to_survey(incoming, aoi):
+def histogram_to_survey_census(incoming):
```
Comment (rajadain) [inline src/mmw/apps/modeling/tasks.py]: You need to keep two parameters here. The first parameter is automatically supplied by the previous task in the Celery chain. The second argument is the area of interest specified in views.py:306 below.

File: src/mmw/apps/modeling/views.py [hunk 3]
```diff
@@ -303,7 +303,7 @@ def start_analyze(request, format=None):
              .set(exchange=exchange, routing_key=routing_key),
         tasks.get_histogram_job_results.s()
              .set(exchange=exchange, routing_key=routing_key),
-        tasks.histogram_to_survey.s(area_of_interest)
+        tasks.histogram_to_survey_census.s(area_of_interest)
```
Comment (rajadain) [inline src/mmw/apps/modeling/views.py]: Or rather, remove the `area_of_interest` parameter from here.

**Comment by jtarricone** [comment]
My bad, it was a last-minute fix to satisfy the linter, only half the change was committed.

File: src/mmw/js/src/analyze/views.js [hunk 4]
```diff
@@ -237,7 +237,7 @@ var TabContentView = Marionette.LayoutView.extend({
 
     showResults: function() {
         var name = this.model.get('name'),
-            results = JSON.parse(this.model.get('taskRunner').get('result')),
+            results = JSON.parse(this.model.get('taskRunner').get('result')).survey,
```
Comment (rajadain) [inline src/mmw/js/src/analyze/views.js]: While the `analyze` endpoint now returns both census and survey, the other endpoints `analyze/animals` and `analyze/pointsources` don't, causing those tables in the Analyze tab to be empty even when there is real data.

**Comment by rajadain** [comment]
If I try and start TR-55 before the Analyze phase completes, the app crashes:

image

This is behavior we have long supported.

File: src/mmw/js/src/modeling/models.js [hunk 5]
```diff
@@ -135,7 +135,8 @@ var ProjectModel = Backbone.Model.extend({
         is_activity: false,             // Project that persists across routes
         gis_data: null,                 // Additionally gathered data, such as MapShed for GWLF-E
         needs_reset: false,             // Should we overwrite project data on next save?
-        allow_save: true                // Is allowed to save to the server - false in compare mode
+        allow_save: true,               // Is allowed to save to the server - false in compare mode
+        aoi_census: null                // JSON blob
```
Comment (rajadain) [inline src/mmw/js/src/modeling/models.js]: Try to phrase the comment to describe what is being stored, rather than how.

File: src/mmw/js/src/modeling/controllers.js (hunk 1 above)
Comment (jtarricone) [inline src/mmw/js/src/modeling/controllers.js]: Good point.

**Comment by jtarricone** [comment]
I committed a couple fixups that should address all points. I'm not sure how necessary it ultimately is, but I also did the round of tests as a logged-in user to make sure project state persisted and launched correctly, which appeared to be the case.

**Comment by rajadain** [comment]
Taking another look.

File: src/mmw/js/src/modeling/controllers.js [hunk 6]
```diff
@@ -114,15 +114,37 @@ var ModelingController = {
             project = App.currentProject;
             project.set('model_package', modelPackage);
             itsiResetProject(project);
+            App.getMapView().addSidebarToggleControl();
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Why is this needed here when it also exists on line 148?

File: src/mmw/js/src/modeling/controllers.js [hunk 7]
```diff
@@ -114,15 +114,37 @@ var ModelingController = {
...
+                area_of_interest: App.map.get('areaOfInterest'),
+                area_of_interest_name: App.map.get('areaOfInterestName'),
+                model_package: modelPackage,
+                scenarios: new models.ScenariosCollection()
+            });
+
             App.currentProject = project;
+            lock.resolve();
+
+            App.getAnalyzeCollection().findWhere({taskName:'analyze'}).fetchAnalysisIfNeeded()
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Since `App.getAnalyzeCollection().findWhere({ taskName: 'analyze' })` is used multiple times, consider assigning it to a variable like `landSoilAnalyzeTask`.

File: src/mmw/js/src/modeling/controllers.js [hunk 8]
```diff
@@ -114,15 +114,37 @@ var ModelingController = {
...
+            .done(function() {
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Consider indenting the `.done` and `.fail` blocks one level to improve readability.

File: src/mmw/js/src/modeling/controllers.js [hunk 9]
```diff
@@ -114,15 +114,37 @@ var ModelingController = {
...
+                App.currentProject.set(
+                    'aoi_census',
+                    JSON.parse(
+                        App.getAnalyzeCollection().findWhere({taskName:'analyze'}).get('result')
+                    ).census
+                );
+            })
+            .fail(function() { projectCleanUp(); });
```
Comment (rajadain) [inline src/mmw/js/src/modeling/controllers.js]: Prefer fully indented blocks, even when the body is a single statement:

```js
.fail(function() {
    projectCleanup();
});
```

Or even better, simply pass the handler directly rather than wrapping it in a function:

```js
.fail(projectCleanup);
```

File: src/mmw/js/src/modeling/controllers.js (hunk 6 above)
Comment (jtarricone) [inline src/mmw/js/src/modeling/controllers.js]: Good catch, I think that's redundant; I tried to totally separate ITSI/non-ITSI blocks once the project state calisthenics started getting awkward, this is a leftover from that.

**Comment by kellyi** [comment]
This still seems to be causing an error returning the data for the analysis results. For AOIs I've checked I'm getting back no data for all the analysis tabs:

screen shot 2016-09-29 at 1 04 05 pm

**Comment by kellyi** [comment]
HM, the above seems to be an issue with my local because it's happening on `develop` too.

**Comment by kellyi** [comment]
Ok, turns out I had not run `./scripts/bundle.sh` on this branch, and I had merged this one into my local `develop`. Checked again on this one after running `bundle` and the results are outputting correctly.

File: src/mmw/js/src/modeling/controllers.js [hunk 10]
```diff
@@ -114,15 +114,37 @@ var ModelingController = {
...
+
             App.currentProject = project;
+            lock.resolve();
+
+            App.getAnalyzeCollection().findWhere({taskName:'analyze'}).fetchAnalysisIfNeeded()
+            .done(function() {
+                App.currentProject.set(
+                    'aoi_census',
+                    JSON.parse(
+                        App.getAnalyzeCollection().findWhere({taskName:'analyze'}).get('result')
```
Comment (kellyi) [inline src/mmw/js/src/modeling/controllers.js]: Would it be worthwhile to indent this like:

```js
App.getAnalyzeCollection()
       .findWhere({...})
       .get(...)
```

**Comment by kellyi** [comment]
+1. Tested this with a variety of AOIs and it worked. 

Looks like the failing CI build's related to the Postgres update; there's a fix for it on `release` but not on `develop` yet, so it might be worth waiting til the release has been made and merged into `develop`, then rebasing these changes on that so that the tests will pass. Here's the error:

```
TASK [azavea.postgresql-support : Install client API libraries for PostgreSQL] *
Thursday 29 September 2016  15:42:13 -0400 (0:00:00.424)       0:00:06.772 **** 
fatal: [services]: FAILED! => {"cache_update_time": 0, "cache_updated": false, "changed": false, "failed": true, "msg": "'/usr/bin/apt-get -y -o \"Dpkg::Options::=--force-confdef\" -o \"Dpkg::Options::=--force-confold\"     install 'libpq-dev=9.5.*.pgdg14.04+1'' failed: E: Version '9.5.*.pgdg14.04+1' for 'libpq-dev' was not found\n", "stderr": "E: Version '9.5.*.pgdg14.04+1' for 'libpq-dev' was not found\n", "stdout": "Reading package lists...\nBuilding dependency tree...\nReading state information...\n", "stdout_lines": ["Reading package lists...", "Building dependency tree...", "Reading state information..."]}
```

**Comment by hectcastro** [comment]
@azavea-bot rebuild
